src/Mapping/ClassMetadata.php

```diff
@@ -2108,13 +2108,12 @@ public function addEntityListener(string $eventName, string $class, string $meth
      * @param DiscriminatorColumnMapping|mixed[]|null $columnDef
      * @psalm-param DiscriminatorColumnMapping|array{
      *     name: string|null,
-     *     fieldName?: string,
-     *     type?: string,
-     *     length?: int,
+     *     fieldName?: string|null,
+     *     type?: string|null,
+     *     length?: int|null,
      *     columnDefinition?: string|null,
      *     enumType?: class-string<BackedEnum>|null,
-     *     options?:array<string,
-     *     mixed>|null
+     *     options?:array<string, mixed>
```

Comment:
null doesn't seems supported.

We pass the value as-is to 
```
$this->discriminatorColumn = DiscriminatorColumnMapping::fromMappingArray($columnDef);
```
which doesn't support null options.

An other option would be to add 
```
if (! isset($columnDef['options'])) {
                $columnDef['options'] = [];
            }
```
if you prefer

Reply:
You've made all other optional keys nullable, yet in this case you opted for the opposite. 🤔 

Let's keep the it nullable for consitency. Not setting the key and setting it to `null` should be semantically equal anyway.

Reply:
Ok, I added the code to support NULL then https://github.com/doctrine/orm/pull/11226/commits/7b776787d72ef887b509a81187a1b83fe05283b0